 fs/uflashfs.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/uflashfs.c b/fs/uflashfs.c
index 330b144..2b61251 100644
--- a/fs/uflashfs.c
+++ b/fs/uflashfs.c
@@ -962,6 +962,7 @@ static int EntryTruncate(UFLASHVOLUME * vol, blknum_t lbe)
     FlashReadEntry(vol->vol_ifc, b_old, &eh, &name);
     FlashWriteBlockHead(vol->vol_ifc, b, &bh);
     FlashWriteEntry(vol->vol_ifc, b, &eh, name);
+    free(name);
     FlashUnitCommit(vol->vol_ifc, b, 0);
 
     /* Update block footer in the last unit. */
